src/models/app_data_json_format.rs

```diff
@@ -116,14 +133,130 @@ mod tests {
                         .unwrap(),
                     version: "6.6.6".to_string(),
                 }),
-                quote: Some(Quote {
+                quote: Some(Quote::V1(QuoteV1 {
                     version: String::from("1.0"),
                     sell_amount: U256::from_dec_str("23426235345").unwrap(),
                     buy_amount: U256::from_dec_str("2341253523453").unwrap(),
+                })),
+            }),
+        };
+
+        assert_eq!(json, expected);
+    }
+    #[test]
+    fn test_loading_json_v4_with_slippage() {
+        let value = json!({
+                "version":"1.2.3",
+                "appCode":"MooSwap",
+                "environment": "production",
+                "metadata":{
+                    "environment": "production",
+                    "referrer":{
+                        "kind":"referrer",
+                        "address":"0x8c35B7eE520277D14af5F6098835A584C337311b",
+                        "version":"6.6.6"
+                },
+                "quote": {
+                    "version": "2.0",
```

Comment:
FYI, we are using [semver](https://semver.org/), thus it has 3 parts.

The latest `quote` metadata version is `0.2.0` rather than `2.0`

I'm sorry if this isn't clear in the schema. I'll get there

Reply:
Thanks for letting me know. Although in this case it doesn't really matter what the version of
Is since it's just a string that gets parsed you'll notice all the version numbers in the other tests here are just 1.2.3

Reply:
How do you map then quote version 0.1.0 and quote version 0.2.0 to their respective types?

Reply:
Serde just parses the content based on what it is.

Reply:
I can change the numbers to 0.2.0 if it makes you feel better though. However this is just a Jain string deserialization into an app data struct. The versions are just strings.

Reply:
Fair enough thanks for explaining